osf/models/mixins.py

```diff
@@ -1961,11 +1961,43 @@ def log_params(self):
     def get_spam_fields(self):
         return NotImplementedError()
 
-    def confirm_spam(self, save=False):
-        super(SpamOverrideMixin, self).confirm_spam(save=False)
+    def confirm_spam(self, save=True):
+        """
+        This should add behavior specific nodes/preprints confirmed to be spam.
+        :param save:
+        :return:
+        """
+        super().confirm_spam(save=save)
+        self.deleted = timezone.now()
         self.set_privacy('private', auth=None, log=False, save=False)
+
+        log = self.add_log(
+            action=self.log_class.MARK_SPAM,
+            params=self.log_params,
+            auth=None,
+            save=False
+        )
+        log.should_hide = True
+        log.save()
+        if save:
+            self.save()
+
+    def confirm_ham(self, save=True):
+        """
+        This should add behavior specific nodes/preprints confirmed to be ham.
+        :param save:
+        :return:
+        """
+        super().confirm_ham()
+
+        if self.logs.filter(action=self.log_class.MARK_SPAM):
+            self.set_privacy('public', log=False)
```

Comment:
Actually, follow-up Q:

For preprints that are flagged during submission without ever being public, and still in the `initial` phase, should this make a best-effort attempt to complete submission?

Reply:
Yeah I'm not sure if there's good answer to that, we could just let the user set the status manually.

Reply:
That would require a "DraftPreprints"-type workflow, as preprints abandoned in the "`initial`" state currently can't be accessed via UI.

Since an actual resolution to this problem should be done in [this ticket](https://openscience.atlassian.net/browse/ENG-1943), could you at least ensure that no preprints (at least, none with `machine_state` other than `accepted`) get set to public here?

Reply:
@Johnetordoff ^^